db/migrations/20230501205418_cache.sql

```diff
@@ -0,0 +1,39 @@
+-- migrate:up
+
+CREATE TABLE IF NOT EXISTS swoop.input_items (
+  item_uuid uuid NOT NULL DEFAULT gen_random_uuid(),
+  item_id text,
+  collection text
+);
+
+CREATE INDEX ON swoop.input_items (item_uuid);
+
+
+CREATE TABLE IF NOT EXISTS swoop.item_payload (
+  item_uuid uuid NOT NULL,
+  payload_uuid uuid NOT NULL
+);
+
+CREATE INDEX ON swoop.item_payload (item_uuid);
+CREATE INDEX ON swoop.item_payload (payload_uuid);
+
+
+CREATE TABLE IF NOT EXISTS swoop.payload_cache (
+  cache_uuid uuid NOT NULL DEFAULT gen_random_uuid(),
+  payload_uuid uuid NOT NULL,
```

Comment:
With the restructuring of the other tables, I think the `payload_uuid` _has_ to be the one that is kept? Because if that is taken out, there is no way to relate the `payload_cache` table with the `item_payload` table.

Reply:
Also, this should be a foreign key to the `payload_uuid` in the `item_payload` table correct?

Reply:
You can rename `payload_uuid` in other tables to match whatever the uuid primary key column is named here. So no, it doesn't have to be `payload_uuid`, unless you don't update those references.

It cannot be a foreign key to the `item_payload` table because the relation between `payload_cache` and `item_payload` is one-to-many. And you'll end up with a situation where you cannot insert a payload: as the primary key on this table, a non-null value is required. But the reference in `item_payload` is also required, so you end up with a situation where you can't insert into either table.